app/src/screen-components/user-news/news-filter.tsx

```diff
@@ -1,190 +1,241 @@
-import React, { useState, useEffect, useRef, FC } from "react"
-import { MaterialCommunityIcons } from "@expo/vector-icons"
-import useFonts from "../../theme/fonts"
-import { useTheme } from '../../theme'
-import CategoryButton from "../../components/buttons/category-button"
-import { Text, View, StyleSheet, ScrollView, Animated, TouchableWithoutFeedback, Easing } from "react-native"
-import categories from "../../shared/categories"
+import React, { useState, useEffect, useRef, FC } from "react";
+import { MaterialCommunityIcons } from "@expo/vector-icons";
+import useFonts from "../../theme/fonts";
+import { useTheme } from "../../theme";
+import CategoryButton from "../../components/buttons/category-button";
+import {
+  Text,
+  View,
+  StyleSheet,
+  ScrollView,
+  Animated,
+  TouchableWithoutFeedback,
+  Easing,
+} from "react-native";
+import categories from "../../shared/categories";
 
 export enum TimeRangeItem {
-    now,
-    today,
-    week,
-    month
+  now,
+  today,
+  week,
+  month,
 }
 
 export interface TimeRange {
-    gteDate?: Date
-    lteDate?: Date
+  gteDate?: Date;
+  lteDate?: Date;
 }
 
 const timeRanges = [
-    { id: TimeRangeItem.now, name: "Nyní" },
-    { id: TimeRangeItem.today, name: "Dnes" },
-    { id: TimeRangeItem.week, name: "Tento týden" },
-    { id: TimeRangeItem.month, name: "Tento měsíc" }
-]
+  { id: TimeRangeItem.now, name: "Nyní" },
+  { id: TimeRangeItem.today, name: "Dnes" },
+  { id: TimeRangeItem.week, name: "Tento týden" },
+  { id: TimeRangeItem.month, name: "Tento měsíc" },
+];
 
-type FilterChageCallback = (selectedCategories: string[], selectedTimeRange: TimeRangeItem) => void
+type FilterChageCallback = (
+  selectedCategories: string[],
+  selectedTimeRange: TimeRangeItem
+) => void;
 
 interface INewsFilter {
-    initialTimeRange: TimeRangeItem
-    initialCategories: string[]
-    isExpanded: boolean
-    onFilterChange: FilterChageCallback
-    onExpadedChange: (isExpanded: boolean) => void
+  initialTimeRange: TimeRangeItem;
+  initialCategories: string[];
+  isExpanded: boolean;
+  onFilterChange: FilterChageCallback;
+  onExpadedChange: (isExpanded: boolean) => void;
 }
 
-const NewsFilter: FC<INewsFilter> = ({ initialTimeRange, initialCategories, isExpanded, onFilterChange, onExpadedChange }) => {
-    const theme = useTheme()
-    const fonts = useFonts()
-
-    const [selectedTimeRange, setTimeRange] = useState(initialTimeRange)
-    const [selectedCategories, setCategories] = useState(initialCategories)
-
-    const [expanded, setExpanded] = useState(isExpanded)
-    const animationHeight = useRef(new Animated.Value(0)).current
-    const animationOpacity = useRef(new Animated.Value(0)).current
-
-    const toggleExpansion = () => {
-        setExpanded(!expanded)
-        onExpadedChange(expanded)
+const NewsFilter: FC<INewsFilter> = ({
```

Comment:
dle konvence jsou props pro danout komponentu pojmenované *Component*Props, tedy místo `INewsFilter` by bylo lepší `NewsFilterProps`... a dal bych `export interface NewsFilterProps`, ať s tím můžou další komponety kdyžtak pracovat.

Reply:
mrknu na to, dik za info